File: Data Structures & Algorithms/reverse-integer/submission-2.py
class Solution:
    def reverse(self, x: int) -> int:
        neg = (x < 0)
        out = 0

        # MAX = 2**31 - 1

        x = abs(x)
        while x!=0 and out <= 2**31 - 1:
            out *= 10
            out += x%10
            x //= 10
        
        if out > 2**31 - 1:
            return 0

        if neg: return -out
        else: return out

File: Data Structures & Algorithms/longest-continuous-subarray-with-absolute-diff-less-than-or-equal-to-limit/submission-2.cpp
#include<queue>

class Solution {
public:
    int longestSubarray(vector<int>& nums, int limit) {
        deque<pair<int,int>> mq;
        deque<pair<int,int>> hq;

        int maxd = 0;

        int l = 0;
        for (int r = 0; r < nums.size(); ++r ) {
            while (!mq.empty() && mq.back().first >= nums[r]) {mq.pop_back();}
            mq.push_back({nums[r], r});
            while (!hq.empty() && hq.back().first <= nums[r]) {hq.pop_back();}
            hq.push_back({nums[r], r}); // r is necessarily in array

            while (hq.front().first - mq.front().first > limit) {
                if (l >= mq.front().second) {mq.pop_front();}
                if (l >= hq.front().second) {hq.pop_front();}
                ++l;
            } // l excluded

            maxd = max(maxd, r-l+1);
        }
        return maxd;
        
    }
};